fixit/cli/__init__.py

```diff
@@ -239,4 +237,26 @@ def ipc_main(opts: LintOpts) -> IPCResult:
         for result in results:
             print(result)
 
-    return IPCResult(args.paths)
+    return IPCResult(list(paths))
+
+
+def ipc_main(opts: LintOpts) -> IPCResult:
+    """
+    Like `run_ipc` instead this function expects arguments to be collected through argparse.
+    This IPC helper takes paths of source files from either a path file (with @paths arg)
+    or a list of paths as args.
+
+    Returns an IPCResult object.
+    """
+    parser = argparse.ArgumentParser(
+        description="Runs Fixit lint rules and print results as console output.",
+        fromfile_prefix_chars="@",
+        parents=[get_multiprocessing_parser()],
+    )
+    parser.add_argument("paths", nargs="*", help="List of paths to run lint rules on.")
+    parser.add_argument("--prefix", help="A prefix to be added to all paths.")
+    args: argparse.Namespace = parser.parse_args()
+
+    return run_ipc(
+        opts=opts, paths=args.paths, prefix=args.prefix, workers=args.workers
+    )
```

Comment:
Let's deprecate this.

Reply:
👍 Added a deprecation warning using python's standard `warnings.warn` functionality.